### PR TITLE
Fas 16 get customer should return customer locations too

### PR DESCRIPTION
**Pull request by NeverlandMJ**

### CODE REVIEW

File: src/api_gateway/models/customer.go [hunk 1]
```diff
@@ -51,6 +51,10 @@ func (c *CustomerCreateReq) Validate() error {
 		return constants.ErrIncorrectNameValue
 	}
 
+	if !libsUtils.IsPhoneValid(c.PhoneNumber) {
```
Comment (khdoba2000) [inline src/api_gateway/models/customer.go:54]: As I see, function libsUtils.IsPhoneValid is using incorrect regex "var phoneRegex = regexp.MustCompile(`(0|\\+62|062|62)[0-9]+$`)"   please fix this regex. There is correct one just below this lines.

File: src/customer_service/pkg/mappers/mapper.go [hunk 2]
```diff
@@ -99,17 +74,19 @@ func OpretorNewCustomer(req *pb.OperatorCreateCustomerRequest) entity.Customer {
...
-			CustomerID:      id.String(),
-			AddressLine1:    req.AddressLine1,
-			AddressLine2:    req.AddressLine2,
-			ApartmentNumber: req.ApartmentNumber,
-			Floor:           req.Floor,
-			Entrance:        req.Entrance,
-			Notes:           req.Notes,
-			Latitude:        req.Latitude,
-			Longitude:       req.Longitude,
+		LocationInfo: []entity.Location{
```
Comment (khdoba2000) [inline src/customer_service/pkg/mappers/mapper.go:77]: why did you change from single Location to array of locations?

Comment (khdoba2000) [reply src/customer_service/pkg/mappers/mapper.go:77]: when creating customer it is ok to register only a single location

File: src/api_gateway/models/customer.go (hunk 1 above)
Comment (NeverlandMJ) [reply src/api_gateway/models/customer.go:54]: that one didn't work correctly for example when i entered phone number "+99856678" it accepted as correct phone number

Comment (khdoba2000) [reply src/api_gateway/models/customer.go:54]: Yeah it was intended to accept numbers globally that means minimum of ~5-6 digits is OK. Just create another REGEX for UZB like [+]{1}[99]{1}[0-9]{10}$ you can check this regex on https://regex101.com/

File: src/customer_service/storage/postgres/customer.go [hunk 3]
```diff
@@ -118,13 +118,23 @@ func (r *customerRepo) EditProfile(ctx context.Context, new entity.UpdateCustome
 
 func (r *customerRepo) GetCustomerByPhoneNumber(ctx context.Context, phone string) (entity.DBCustomer, error) {
 	var customer entity.DBCustomer
-	err := r.db.GetContext(ctx, &customer, `SELECT * FROM customer WHERE phone_number=$1`, phone)
+	err := r.db.GetContext(ctx, &customer,
```
Comment (khdoba2000) [inline src/customer_service/storage/postgres/customer.go:121]: again question about consistency, wont we have some problems when the model customer is updated? It is better to use QueryRowContext for single row expected queries(or QueryContext for multiple rows), scanning just the fields the endpoint needs.

Comment (NeverlandMJ) [reply src/customer_service/storage/postgres/customer.go:121]: get context is faster and inside it uses query row context anyways

Comment (khdoba2000) [reply src/customer_service/storage/postgres/customer.go:121]: how can it be faster if it uses QueryRowCtx inside? Please note that I am afraid any changes done to the model(like adding or removing a field) breaks this query, it is bad.

File: src/customer_service/storage/postgres/customer.go [hunk 4]
```diff
@@ -143,20 +179,54 @@ func (r *customerRepo) GetCustomerByPhoneNumber(ctx context.Context, phone strin
...
+				&l.AddressLine2,
+				&l.ApartmentNumber,
+				&l.Floor,
+				&l.Entrance,
+				&l.Notes,
+				&l.CreatedAt,
+				&l.UpdatedAt,
+			)
+			if err != nil {
+				log.Println("getting location info: ", err)
```
Comment (khdoba2000) [inline src/customer_service/storage/postgres/customer.go:218]: why you are just logging errors? you should return them.

File: src/customer_service/storage/postgres/customer.go [hunk 5]
```diff
@@ -143,20 +179,54 @@ func (r *customerRepo) GetCustomerByPhoneNumber(ctx context.Context, phone strin
...
+			}
+			locations = append(locations, l)
+		}
 		customer.LocationInfo = locations
 	}
 
+	err = rows.Err()
+	if err != nil {
+		log.Println("getting location info: ", err)
```
Comment (khdoba2000) [inline src/customer_service/storage/postgres/customer.go:227]: here too

**Review by khdoba2000 — CHANGES_REQUESTED** [review]

**Review by khdoba2000 — APPROVED** [review]
